tests/test_builders/test_build_warnings.py

```diff
@@ -58,6 +59,21 @@ def test_html_warnings(app):
     _check_warnings(warnings_exp, app.warning.getvalue())
 
 
+@pytest.mark.parametrize('pdb', [True, False])
+@pytest.mark.sphinx('html', testroot='warnings', freshenv=True)
+def test_html_warnings_pdb(app, pdb):
+    app.pdb = pdb
+    app.warningiserror = True
+    try:
+        app.build(force_all=True)
+    except Exception:
+        tb = traceback.format_exc()
+        if pdb:
+            assert "unindent_warning" in tb
```

Comment:
Just wondering but should the `unindent_warning` only be present when pdb is set? if so I would add an `assert not in` in the else body (same question for the`pending_warning`)